taichi/backends/cuda/codegen_cuda.cpp

```diff
@@ -105,41 +105,40 @@ class CodeGenLLVMCUDA : public CodeGenLLVM {
   }
 
   void visit(PrintStmt *stmt) override {
+    // TODO(@archibate): ask @yuanming-hu about how to deal with CUDA print
+
     TI_ASSERT(stmt->width() == 1);
 
-    auto value_type = tlctx->get_data_type(stmt->stmt->ret_type.data_type);
+    std::vector<llvm::Type *> types;
+    std::vector<llvm::Value *> values;
 
-    std::string format;
+    std::string formats;
+    for (auto content : stmt->contents) {
+      if (formats.size() != 0)  // not the first expr?
```

Comment:
OK, since you did use `fmt::join` (thx!), could you also do the same here?

Reply:
No, this is not the case. We will have to use `fmt::format("[debug] {}\n", fmt::join(formats))` for this, while we already do the format works in `lang_util.h:data_type_format()` and `vprintf` at runtime.

Reply:
Sorry I don't get your point. What's the problem with using `fmt::format("[debug] {}\n", fmt::join(formats, " "))`?

Reply:
Overkill.

Reply:
> Sorry I don't get your point. What's the problem with using `fmt::format("[debug] {}\n", fmt::join(formats, " "))`?

`data_type_format(content->ret_type.data_type) for content in stmt->contents` doesn't form an `std::vector`.

Reply:
We could just change `formats` to `std::vector<std::string>`..